app/javascript/components/firmware-registry-add-edit.jsx

```diff
@@ -0,0 +1,135 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Grid, Row } from 'patternfly-react';
+import { componentTypes, validatorTypes } from '@data-driven-forms/react-form-renderer';
+import MiqFormRenderer from '../forms/data-driven-form';
+
+const FirmwareRegistryAddEdit = (props) => {
+  const TYPES = [
+    ['FirmwareRegistry::RestApiDepot', __('Rest API Depot')],
+  ];
+
+  const [initialValues, setInitialValues] = useState({ type: TYPES[0][0] });
+
+  useEffect(() => {
+    props.dispatch({
+      type: 'FormButtons.init',
+      payload: {
+        newRecord: true,
+        pristine: true,
+      },
+    });
+    props.dispatch({
+      type: 'FormButtons.customLabel',
+      payload: __('Add'),
+    });
+  }, []);
+
+  const submitValues = (values) => {
+    API.post('/api/firmware_registries', values).then((response) => {
+      if (response.results) { add_flash(__('Firmware Registry added sucessfully', 'success')); } else { add_flash(response.error.message, 'error'); }
+    });
+  };
+
+  const handleFormStateUpdate = (formState) => {
+    props.dispatch({
+      type: 'FormButtons.saveable',
+      payload: formState.valid,
+    });
+    props.dispatch({
+      type: 'FormButtons.pristine',
+      payload: formState.pristine,
+    });
+    props.dispatch({
+      type: 'FormButtons.callbacks',
+      payload: { addClicked: () => submitValues(formState.values) },
+    });
+  };
+
+  return (
+    <Grid fluid>
+      <Row>
+        <MiqFormRenderer
+          initialValues={initialValues}
+          schema={
```

Comment:
For schema we usually create separate file like `some-component.schema.js`. It will be ignored by linters.

Also the component and schema and all other related files should be placed in separate folder. (Check some existing forms)